File: MarketMakerAPI/Models/Transaction.cs
namespace MarketMaker.Models;

public record Transaction(
    string BuyerUser,
    Guid BuyerOrderId,
    string SellerUser,
    Guid SellerOrderId,
    string Market,
    long Price,
    int Quantity,
    Guid PassiveOrder,
    DateTime TimeStamp 
);

File: NewFrontend/client/transaction.js
export class Transaction{
    constructor(transaction){
        this.buyerUser = transaction.buyerUser;
        this.buyerOrderId = transaction.buyerOrderId;
        this.sellerUser = transaction.sellerUser;
        this.sellerOrderId = transaction.sellerOrderId;
        this.price = transaction.price;
        this.quantity = transaction.quantity;
        if(transaction.passiveOrder = transaction.buyerOrderId){
            this.passiveUser = this.buyerUser;
            this.aggressiveUser = this.sellerUser;
        }else{
            this.passiveUser = this.sellerUser;
            this.aggressiveUser = this.buyerUser;
        }
        this.timeStamp = transaction.timeStamp;
    }
}